web/test/pages/profile/profile-shared.test.tsx

```diff
@@ -271,4 +310,53 @@ describe("profile - shared", () => {
       expect(screen.queryByTestId("opp-alert")).not.toBeInTheDocument();
     });
   });
+
+  describe("non essential questions", () => {
+    const generateBusinessForNonEssentialQuestionTest = (profileData: Partial<ProfileData>): Business => {
+      return generateBusiness({
+        profileData: generateProfileData({
+          ...profileData,
+          foreignBusinessType: profileData.businessPersona === "FOREIGN" ? "NEXUS" : undefined,
+          nexusLocationInNewJersey: profileData.businessPersona === "FOREIGN" ? false : undefined,
+          foreignBusinessTypeIds:
+            profileData.businessPersona === "FOREIGN" ? ["employeeOrContractorInNJ"] : undefined,
+        }),
+      });
+    };
+
+    it.each(["STARTING", "FOREIGN"])(
+      "adds non essential questions if industry has them when %s",
+      async (businessPersona) => {
+        const business = generateBusinessForNonEssentialQuestionTest({
+          industryId: "test-industry-2",
+          businessPersona: businessPersona as BusinessPersona,
+        });
+        renderPage({ business, setRegistrationModalIsVisible });
+        selectByValue("Industry", "test-industry");
+        clickSave();
+        await waitFor(() => {
+          expect(currentBusiness().profileData.nonEssentialQuestions).toStrictEqual({
+            "test-question-1": false,
+            "test-question-2": false,
+          });
```

Comment:
Would it be good to have a test here where the questions are true? Like the non essential question exist / are true

Reply:
I didn't add that test in this file because the `true` and `false` accurately being updated is being tested at the component level in `ProfileNonEssentialQuestion.test.tsx`?